addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -0,0 +1,320 @@
+            let $commonColorSection = this.$('[data-name="common"]');
+            summernoteCustomColors.forEach((colorRow, i) => {
+                if (i === 0) {
+                    return;  // Ignore the summernote gray palette and use ours
+                }
+                let $div = $('<div/>', {class: 'clearfix'}).appendTo($commonColorSection);
+                colorRow.forEach(color => {
+                    $div.append(this._createColorButton(color, ['o_common_color']));
+                });
+            });
+        }
+
+        // Render custom colors
+        if (!this.options.excluded.includes('custom')) {
+            const $editable = window.__EditorMenuBar_$editable || $();
+            const customColors = _.chain($editable.find('[style*="color"]')).map(function (el) {
+                if (el.style.color) {
+                    return el.style.color;
+                }
+                if (el.style.backgroundColor) {
+                return el.style.backgroundColor;
+                }
+            }).uniq().value();
+            if (customColors.length) {
+                customColors.forEach(color => {
+                    if (!this.summernoteCustomColorsArray.includes(color)) {
+                        this._addCustomColorButton(color);
+                    }
+                });
+            }
+        }
+
+        // TODO refactor in master
+        // The primary and secondary are hardcoded here (but marked as hidden)
+        // so they can be removed from snippets when selecting another color.
+        // Normally, the chosable colors do not contain them, which prevents
+        // them to be removed. Indeed, normally, the 'alpha' and 'beta' colors
+        // (which are the same) are displayed instead... but not for all themes.
+        this.$el.append($('<button/>', {'class': 'd-none', 'data-color': 'primary'}));
+        this.$el.append($('<button/>', {'class': 'd-none', 'data-color': 'secondary'}));
+
+        // Compute class colors
+        this.classes = [];
+        this.el.querySelectorAll('button[data-color]').forEach(elem => {
+            const color = elem.dataset.color
+            const $color = $(elem);
+            const className = 'bg-' + color;
+            $color.addClass(className);
+            if (this.options.$target && this.options.$target.hasClass(className)) {
+                $color.addClass('selected');
+            }
+            this.classes.push(className);
+        });
+
+        return res;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Add a custom button in the coresponding section.
+     *
+     * @private
+     * @param {string} color Css rgb(a) string
+     * @param {Array<string>} classes Classes added to the button
+     */
+    _addCustomColorButton: function (color, classes = []) {
```

Comment:
Default value for parameters: unknown support for safari, could you check ? :)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/default_parameters#Browser_compatibility

Reply:
Mmh, I only looked at caniuse and only found "Destructured parameter with default value assignment", ok then :+1: